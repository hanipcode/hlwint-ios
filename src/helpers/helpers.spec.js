import * as service from './api';
import { generateStreamLink } from '../data/wowza';

if (!global.fetch) {
  global.fetch = require('node-fetch'); // eslint-disable-line global-require
}
jest.setTimeout(100000);

describe('Hlwint API Service', () => {
  let data;
  let liveList;
  beforeEach(async () => {
    const currentUser = {
      firstName: 'Ko',
      lastName: 'Kyaw',
      gender: 'male',
      dateOfBirth: null,
      email: null,
      phoneNumber: '08',
      profilePic: 'https://graph.facebook.com/367599367030660/picture?type=large',
      facebookId: '367599367030660',
    };
    data = await service.facebookLogin(
      currentUser.firstName,
      currentUser.lastName,
      currentUser.dateOfBirth,
      currentUser.gender,
      currentUser.email,
      currentUser.phoneNumber,
      currentUser.profilePic,
      currentUser.facebookId,
    );
  });
  it('have a correct login service', async () => {
    const blob = await data.json();
    expect(data.status).toEqual(200);
    expect(blob.message).toEqual('success');
    expect(blob).toMatchSnapshot();
  });
  it('have correct data', async () => {
    const blob = await data.json();
    expect(blob.data.u_token.length).toBeGreaterThan(0);
  });

  it('able to upload last device', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const lastDevice = 'android test';
    const deviceOS = 'android';
    const uploadDeviceService = await service.uploadLastDevice(id, lastDevice, deviceOS, u_token);
    const uploadDeviceData = await uploadDeviceService.json();
    expect(uploadDeviceData.message).toEqual('success');
    expect(uploadDeviceService.status).toEqual(200);
    expect(uploadDeviceData).toMatchSnapshot();
  });

  it('able to get current user profile', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const getProfileData = await service.getUserProfile(id, u_token);
    expect(getProfileData.message).toEqual('success');
    expect(getProfileData).toMatchSnapshot();
  });

  it('able to follow user', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const doFollowUser = await service.addOrRemoveFollowing(id, u_token, 43);
    expect(doFollowUser.message).toMatch(/success/);
    expect(doFollowUser).toMatchSnapshot();
  });

  it('ablle to get user following', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const userFollowing = await service.getUserFollowing(id, u_token);
    expect(userFollowing.message).toEqual('success');
    expect(userFollowing).toMatchSnapshot();
  });

  it('able to get user follower', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const userFollower = await service.getUserFollower(id, u_token);
    expect(userFollower.message).toEqual('success');
    expect(userFollower).toMatchSnapshot();
  });

  it('able to input task to task log', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const type = 'API_TEST';
    const parameter = null;
    const sendTask = await service.sendTask(id, type, parameter, u_token);
    const sendTaskData = await sendTask.json();
    expect(sendTask.status).toEqual(200);
    expect(sendTaskData.message).toEqual('success');
    expect(sendTaskData).toMatchSnapshot();
  });

  it('able to retrieve live data', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const liveRetrieveData = await service.liveDataRetrieval(id, u_token);
    expect(liveRetrieveData.message).toEqual('success');
    expect(liveRetrieveData).toMatchSnapshot();
    expect(liveRetrieveData.live_lists).toMatchSnapshot();
    liveList = liveRetrieveData;
  });

  it('able to get explore data', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const exploreData = await service.getExploreData(id, u_token);
    expect(exploreData.message).toEqual('success');
    expect(exploreData).toMatchSnapshot();
  });

  it('able to get Top Ranked user', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const topRankData = await service.getTopRankData(id, u_token, 10, 10);
    expect(topRankData.message).toEqual('success');
    expect(topRankData).toMatchSnapshot();
  });

  it('able to get nearby user', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const nearbyData = await service.getNearbyData(id, u_token, 10, 10);
    expect(nearbyData.message).toEqual('success');
    expect(nearbyData).toMatchSnapshot();
  });

  it('able to get Broadcast Detail', async () => {
    if (liveList.length === 0) {
      console.warn("Can't test broadcast detail, no available broadcast");
      return;
    }
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const userStream = liveList.live_lists[0];
    const broadcastDetailData = await service.getBroadcastDetail(id, u_token, userStream.id);
    const { hash, b_user_id } = broadcastDetailData.data;
    console.log(generateStreamLink(hash, b_user_id));
    expect(broadcastDetailData).toMatchSnapshot();
  });

  it('able to list tags', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const tagList = await service.getTagList(id, u_token);
    expect(tagList.message).toEqual('success');
    expect(tagList).toMatchSnapshot();
  });

  it('able to create tag', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const createTagData = await service.createTag(id, u_token, 'TESTAG');
    expect(createTagData.message).toEqual('success');
    expect(createTagData).toMatchSnapshot();
  });

  it('able to create broadcast', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const title = 'test title';
    const tags = JSON.stringify({ tags: [{ id: 1 }, { id: 2 }] });
    const createBroadcastData = await service.initBroadcast(id, u_token, title, tags);
    expect(createBroadcastData).toMatchSnapshot();
    expect(createBroadcastData.message).toEqual('success');
  });

  it('able to get stream by used tags', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const streamByTags = await service.getStreamByTags(id, u_token, 1);
    expect(streamByTags.message).toEqual('success');
    expect(streamByTags).toMatchSnapshot();
  });

  it('able to join stream', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const joinData = await service.joinLive(id, u_token, id);
    expect(joinData).toMatchSnapshot();
    expect(joinData.message).toEqual('success');
  });
  it('able to get viewer list', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const viewerListData = await service.getViewerList(id);
    expect(viewerListData).toMatchSnapshot();
    expect(viewerListData.message).toEqual('success');
  });
  it('able to send gift', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const sendCashData = await service.sendGift(id, u_token, id, 43, 37, 9, 100000);
    expect(sendCashData).toMatchSnapshot();
    expect(sendCashData.message).toEqual('success');
  });
  it('able to get viewer ended', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const viewerEndedData = await service.getViewerLiveEnded(id, u_token, id);
    expect(viewerEndedData).toMatchSnapshot();
    expect(viewerEndedData.message).toEqual('success');
  });
  it('able to leave the stream', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const leaveStreamData = await service.leaveLive(id, u_token, id);
    expect(leaveStreamData).toMatchSnapshot();
    expect(leaveStreamData.message).toEqual('success');
  });
  it('able to modify end time', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const modifyEndTime = await service.updateBroadcastEndtime(id);
    expect(modifyEndTime.message).toEqual('success');
    expect(modifyEndTime).toMatchSnapshot();
  });
  it('able to get broadcaster ended data', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const broadcasterEndedData = await service.getBroadcasterLiveEnded(id, u_token);
    expect(broadcasterEndedData).toMatchSnapshot();
    expect(broadcasterEndedData.message).toEqual('success');
  });

  it('able to end broadcast', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const destroyBroadcastData = await service.endBroadcast(id, u_token);
    expect(destroyBroadcastData).toMatchSnapshot();
    expect(destroyBroadcastData.message).toEqual('success');
  });

  it('able to get top fan', async () => {
    const blob = await data.json();
    const { id, u_token } = blob.data;
    const topFanData = await service.getTopFan(id);
    expect(topFanData).toMatchSnapshot();
    expect(topFanData.message).toEqual('success');
  }
  /* DIsable for testing because very resourceful */
  // it('able to get server info', async () => {
  //   const serverInfoData = await service.getServerInfo();
  //   expect(serverInfoData).toMatchSnapshot();
  // });

  // it('able to get list of gift', async () => {
  //   const blob = await data.json();
  //   const { id, u_token } = blob.data;
  //   const giftInfoData = await service.getGiftList(id, u_token);
  //   expect(giftInfoData).toMatchSnapshot();
  // });
});
