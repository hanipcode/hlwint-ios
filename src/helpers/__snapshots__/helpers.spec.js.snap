// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hlwint API Service able to create broadcast 1`] = `
Object {
  "data": Object {
    "b_start_time": "2018-01-24 11:26:44",
    "b_title": "test title",
    "b_user_id": 384,
    "hash": "7eb15e1d702fb1d6c124fb495076ce5b",
    "id": 224,
    "rviewers": 0,
    "tviewers": 0,
    "viewers": 0,
  },
  "message": "success",
}
`;

exports[`Hlwint API Service able to create tag 1`] = `
Object {
  "data": Object {
    "id": 207,
    "t_color_code": "#c095de",
    "t_count": 16,
    "t_name": "TESTAG",
    "t_status": "active",
  },
  "message": "success",
}
`;

exports[`Hlwint API Service able to end broadcast 1`] = `
Object {
  "message": "success",
}
`;

exports[`Hlwint API Service able to follow user 1`] = `
Object {
  "message": "delete success",
}
`;

exports[`Hlwint API Service able to get Broadcast Detail 1`] = `
Object {
  "data": Object {
    "b_end_time": null,
    "b_start_time": "2018-01-24 11:22:15",
    "b_title": "hello",
    "b_user_id": 70,
    "hash": "fd0c3fcb267ff46f9568140c2fc9d1dd",
    "id": 169,
    "rviewers": 0,
    "tviewers": 0,
    "viewers": 0,
  },
  "message": "success",
}
`;

exports[`Hlwint API Service able to get Top Ranked user 1`] = `
Object {
  "message": "success",
  "toprank": Array [
    Object {
      "id": 2498,
      "live": 0,
      "rank": "254",
      "u_first_name": "Thun",
      "u_follower_count": 365,
      "u_last_name": "Zaw",
      "u_nick_name": "thun2498",
      "u_profile_pic": "https://graph.facebook.com/1882290015114642/picture?type=large",
    },
    Object {
      "id": 5549,
      "live": 0,
      "rank": "230",
      "u_first_name": "Nang",
      "u_follower_count": 480,
      "u_last_name": "Sein",
      "u_nick_name": "nansandar5549",
      "u_profile_pic": "https://graph.facebook.com/464218360641334/picture?type=large",
    },
    Object {
      "id": 7416,
      "live": 0,
      "rank": "145",
      "u_first_name": "ဟန္",
      "u_follower_count": 133,
      "u_last_name": "ထူးစ်ာန္",
      "u_nick_name": "ဟန္7416",
      "u_profile_pic": "https://graph.facebook.com/1669030049827418/picture?type=large",
    },
    Object {
      "id": 7942,
      "live": 0,
      "rank": "145",
      "u_first_name": "Aung",
      "u_follower_count": 159,
      "u_last_name": "Khant",
      "u_nick_name": "aung7942",
      "u_profile_pic": "https://graph.facebook.com/1697626750267915/picture?type=large",
    },
    Object {
      "id": 1876,
      "live": 0,
      "rank": "115",
      "u_first_name": "Pann",
      "u_follower_count": 222,
      "u_last_name": "Thaw Tar",
      "u_nick_name": "pann1876",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/16_20_301876_profile.png",
    },
    Object {
      "id": 8399,
      "live": 0,
      "rank": "115",
      "u_first_name": "Su",
      "u_follower_count": 308,
      "u_last_name": "Win",
      "u_nick_name": "su8399",
      "u_profile_pic": "https://graph.facebook.com/2117869858499500/picture?type=large",
    },
    Object {
      "id": 2353,
      "live": 0,
      "rank": "115",
      "u_first_name": "April",
      "u_follower_count": 309,
      "u_last_name": "Pink",
      "u_nick_name": "april2353",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/00_24_082353_profile.png",
    },
    Object {
      "id": 1768,
      "live": 0,
      "rank": "110",
      "u_first_name": "Khin",
      "u_follower_count": 1626,
      "u_last_name": "Tar San",
      "u_nick_name": "khin1768",
      "u_profile_pic": "https://graph.facebook.com/1737245439621574/picture?type=large",
    },
    Object {
      "id": 7414,
      "live": 0,
      "rank": "95",
      "u_first_name": "C",
      "u_follower_count": 86,
      "u_last_name": "Win",
      "u_nick_name": "c7414",
      "u_profile_pic": "https://graph.facebook.com/1021888681285769/picture?type=large",
    },
    Object {
      "id": 1888,
      "live": 0,
      "rank": "85",
      "u_first_name": "May",
      "u_follower_count": 349,
      "u_last_name": "Nanda",
      "u_nick_name": "may1888",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/11_27_411888_profile.png",
    },
  ],
}
`;

exports[`Hlwint API Service able to get broadcaster ended data 1`] = `
Object {
  "data": Object {
    "duration": "00:02:08",
    "first_name": "Ko",
    "id": 384,
    "last_name": "Kyaw",
    "new_follower": 0,
    "new_income": 0,
    "nick_name": "ko384",
    "profile_pic": "https://graph.facebook.com/367599367030660/picture?type=large",
    "tag": Array [
      Object {
        "id": 1,
        "t_color_code": "#CCAAEE",
        "t_name": "#beautytips",
      },
      Object {
        "id": 2,
        "t_color_code": "#958882",
        "t_name": "#singing",
      },
    ],
    "title": "test title",
    "total_view": 14,
  },
  "message": "success",
}
`;

exports[`Hlwint API Service able to get current user profile 1`] = `
Object {
  "data": Object {
    "bot": 0,
    "data_usage": null,
    "id": 384,
    "u_account_status": "active",
    "u_biography": null,
    "u_city": "Pathein",
    "u_coins": 13630,
    "u_created_time": "2017-11-13 15:36:22",
    "u_date_of_birth": "null",
    "u_device_os": "android",
    "u_duration": "00:00:00",
    "u_email": "null",
    "u_facebook_id": "367599367030660",
    "u_fan_count": 1,
    "u_first_name": "Ko",
    "u_follower_count": 1,
    "u_full_name": "Ko Kyaw",
    "u_gender": "female",
    "u_google_id": "null",
    "u_income": 0,
    "u_ip_address": "119.2.52.244",
    "u_key_id": "91600",
    "u_larger_profile_pic": null,
    "u_last_device": "android test",
    "u_last_login_time": "2018-01-24 11:25:59",
    "u_last_name": "Kyaw",
    "u_nick_name": "ko384",
    "u_phone_number": "09",
    "u_profile_pic": "https://graph.facebook.com/367599367030660/picture?type=large",
    "u_reason_banned": "",
    "u_reward_status": 0,
    "u_token": "iPccDeXSVePNATDFlV334A8bT434vcnqvEO0RVYx4N5uM19d9xRBj9pWjzN8",
    "u_type": 1,
  },
  "message": "success",
}
`;

exports[`Hlwint API Service able to get explore data 1`] = `
Object {
  "image": "https://dashboard.imliveapp.com/public/mlive_img/mlive.png",
  "message": "success",
  "nearby": Array [
    Object {
      "id": 3018,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/1583112471734457/picture?type=large",
    },
    Object {
      "id": 4112,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/1668681356530530/picture?type=large",
    },
    Object {
      "id": 1768,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/1737245439621574/picture?type=large",
    },
    Object {
      "id": 3194,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/2018473215104706/picture?type=large",
    },
    Object {
      "id": 2085,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/1972688356314634/picture?type=large",
    },
    Object {
      "id": 2494,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/460452244350459/picture?type=large",
    },
    Object {
      "id": 16,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/1602460376480950/picture?type=large",
    },
    Object {
      "id": 7397,
      "live": 0,
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/16_46_397397_profile.png",
    },
    Object {
      "id": 52,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/362155060911834/picture?type=large",
    },
    Object {
      "id": 27,
      "live": 0,
      "u_profile_pic": "https://graph.facebook.com/395585250857991/picture?type=large",
    },
  ],
  "tag": Array [
    Object {
      "id": 8,
      "t_color_code": "#975565",
      "t_count": 58,
      "t_name": "#shorttrip",
      "t_status": "active",
    },
    Object {
      "id": 151,
      "t_color_code": "#de6429",
      "t_count": 56,
      "t_name": "#singing",
      "t_status": "active",
    },
    Object {
      "id": 2,
      "t_color_code": "#958882",
      "t_count": 56,
      "t_name": "#singing",
      "t_status": "active",
    },
    Object {
      "id": 4,
      "t_color_code": "#778766",
      "t_count": 55,
      "t_name": "#funtime",
      "t_status": "active",
    },
    Object {
      "id": 24,
      "t_color_code": "#778766",
      "t_count": 54,
      "t_name": "#popmusic",
      "t_status": "active",
    },
    Object {
      "id": 16,
      "t_color_code": "#009848",
      "t_count": 52,
      "t_name": "#party",
      "t_status": "active",
    },
    Object {
      "id": 21,
      "t_color_code": "#009848",
      "t_count": 51,
      "t_name": "#happyweekend",
      "t_status": "active",
    },
    Object {
      "id": 1,
      "t_color_code": "#CCAAEE",
      "t_count": 51,
      "t_name": "#beautytips",
      "t_status": "active",
    },
    Object {
      "id": 7,
      "t_color_code": "#958882",
      "t_count": 50,
      "t_name": "#nightout",
      "t_status": "active",
    },
    Object {
      "id": 18,
      "t_color_code": "#975565",
      "t_count": 50,
      "t_name": "#beyourvalentine",
      "t_status": "active",
    },
  ],
  "toprank": Array [
    Object {
      "id": 384,
      "live": 0,
      "rank": "10000",
      "u_profile_pic": "https://graph.facebook.com/367599367030660/picture?type=large",
    },
    Object {
      "id": 4112,
      "live": 0,
      "rank": "6780",
      "u_profile_pic": "https://graph.facebook.com/1668681356530530/picture?type=large",
    },
    Object {
      "id": 10159,
      "live": 0,
      "rank": "2070",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/14_50_1510159_profile.png",
    },
    Object {
      "id": 10382,
      "live": 0,
      "rank": "1445",
      "u_profile_pic": "https://graph.facebook.com/377486816046924/picture?type=large",
    },
    Object {
      "id": 3018,
      "live": 0,
      "rank": "1336",
      "u_profile_pic": "https://graph.facebook.com/1583112471734457/picture?type=large",
    },
    Object {
      "id": 775,
      "live": 0,
      "rank": "850",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/19_40_04775_profile.png",
    },
    Object {
      "id": 3194,
      "live": 0,
      "rank": "792",
      "u_profile_pic": "https://graph.facebook.com/2018473215104706/picture?type=large",
    },
    Object {
      "id": 7397,
      "live": 0,
      "rank": "575",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/16_46_397397_profile.png",
    },
    Object {
      "id": 2499,
      "live": 0,
      "rank": "320",
      "u_profile_pic": "https://graph.facebook.com/901009120057896/picture?type=large",
    },
    Object {
      "id": 2494,
      "live": 0,
      "rank": "265",
      "u_profile_pic": "https://graph.facebook.com/460452244350459/picture?type=large",
    },
  ],
}
`;

exports[`Hlwint API Service able to get nearby user 1`] = `
Object {
  "data": Array [
    Object {
      "id": 70,
      "live": 1,
      "u_first_name": "yati",
      "u_follower_count": 583,
      "u_last_name": "zin",
      "u_nick_name": "yati21",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/21_18_1770_profile.png",
    },
    Object {
      "id": 1823,
      "live": 0,
      "u_first_name": "Eaint",
      "u_follower_count": 540,
      "u_last_name": "Treasure",
      "u_nick_name": "eaint1823",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/09_54_401823_profile.png",
    },
    Object {
      "id": 2499,
      "live": 0,
      "u_first_name": "PhomeShein",
      "u_follower_count": 535,
      "u_last_name": "Htet",
      "u_nick_name": "phomeshein2499",
      "u_profile_pic": "https://graph.facebook.com/901009120057896/picture?type=large",
    },
    Object {
      "id": 5549,
      "live": 0,
      "u_first_name": "Nang",
      "u_follower_count": 480,
      "u_last_name": "Sein",
      "u_nick_name": "nansandar5549",
      "u_profile_pic": "https://graph.facebook.com/464218360641334/picture?type=large",
    },
    Object {
      "id": 83,
      "live": 0,
      "u_first_name": "ေမသဥၨာ",
      "u_follower_count": 443,
      "u_last_name": "ေက်ာ္",
      "u_nick_name": "​ေမ83",
      "u_profile_pic": "https://graph.facebook.com/513977715626828/picture?type=large",
    },
    Object {
      "id": 1828,
      "live": 0,
      "u_first_name": "Jue Jue",
      "u_follower_count": 422,
      "u_last_name": "Thit Sar",
      "u_nick_name": "juejue1828",
      "u_profile_pic": "https://graph.facebook.com/559327204410431/picture?type=large",
    },
    Object {
      "id": 1974,
      "live": 0,
      "u_first_name": "Paing Oak",
      "u_follower_count": 414,
      "u_last_name": "Soe Aung",
      "u_nick_name": "paingoak1974",
      "u_profile_pic": "https://graph.facebook.com/2024677957775066/picture?type=large",
    },
    Object {
      "id": 1877,
      "live": 0,
      "u_first_name": "Eaindray",
      "u_follower_count": 382,
      "u_last_name": "Chit Su",
      "u_nick_name": "eaindray1877",
      "u_profile_pic": "https://graph.facebook.com/818952491620096/picture?type=large",
    },
    Object {
      "id": 2498,
      "live": 0,
      "u_first_name": "Thun",
      "u_follower_count": 365,
      "u_last_name": "Zaw",
      "u_nick_name": "thun2498",
      "u_profile_pic": "https://graph.facebook.com/1882290015114642/picture?type=large",
    },
    Object {
      "id": 1888,
      "live": 0,
      "u_first_name": "May",
      "u_follower_count": 349,
      "u_last_name": "Nanda",
      "u_nick_name": "may1888",
      "u_profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/11_27_411888_profile.png",
    },
  ],
  "message": "success",
}
`;

exports[`Hlwint API Service able to get stream by used tags 1`] = `
Object {
  "data": Array [
    Object {
      "id": 384,
      "live": 1,
      "u_first_name": "Ko",
      "u_follower_count": 1,
      "u_last_name": "Kyaw",
      "u_nick_name": "ko384",
      "u_profile_pic": "https://graph.facebook.com/367599367030660/picture?type=large",
    },
  ],
  "message": "success",
}
`;

exports[`Hlwint API Service able to get user follower 1`] = `
Object {
  "data": Array [],
  "message": "success",
}
`;

exports[`Hlwint API Service able to get viewer ended 1`] = `
Object {
  "data": Object {
    "biography": null,
    "first_name": "Ko",
    "following": "false",
    "id": 384,
    "last_name": "Kyaw",
    "nick_name": "ko384",
    "profile_pic": "https://graph.facebook.com/367599367030660/picture?type=large",
    "view_count": 15,
  },
  "message": "success",
}
`;

exports[`Hlwint API Service able to get viewer list 1`] = `
Object {
  "broadcaster_coin": 3620,
  "data": Array [
    Object {
      "id": 384,
      "u_first_name": "Ko",
      "u_follower_count": 1,
      "u_last_name": "Kyaw",
      "u_nick_name": "ko384",
      "u_profile_pic": "https://graph.facebook.com/367599367030660/picture?type=large",
    },
  ],
  "message": "success",
}
`;

exports[`Hlwint API Service able to input task to task log 1`] = `
Object {
  "message": "success",
}
`;

exports[`Hlwint API Service able to join stream 1`] = `
Object {
  "message": "success",
  "viewer_count": 1,
}
`;

exports[`Hlwint API Service able to leave the stream 1`] = `
Object {
  "message": "success",
  "viewer_count": 0,
}
`;

exports[`Hlwint API Service able to list tags 1`] = `
Object {
  "data": Array [
    Object {
      "id": 8,
      "t_color_code": "#975565",
      "t_count": 58,
      "t_name": "#shorttrip",
      "t_status": "active",
    },
    Object {
      "id": 2,
      "t_color_code": "#958882",
      "t_count": 56,
      "t_name": "#singing",
      "t_status": "active",
    },
    Object {
      "id": 151,
      "t_color_code": "#de6429",
      "t_count": 56,
      "t_name": "#singing",
      "t_status": "active",
    },
    Object {
      "id": 4,
      "t_color_code": "#778766",
      "t_count": 55,
      "t_name": "#funtime",
      "t_status": "active",
    },
    Object {
      "id": 24,
      "t_color_code": "#778766",
      "t_count": 54,
      "t_name": "#popmusic",
      "t_status": "active",
    },
    Object {
      "id": 16,
      "t_color_code": "#009848",
      "t_count": 52,
      "t_name": "#party",
      "t_status": "active",
    },
    Object {
      "id": 1,
      "t_color_code": "#CCAAEE",
      "t_count": 51,
      "t_name": "#beautytips",
      "t_status": "active",
    },
    Object {
      "id": 21,
      "t_color_code": "#009848",
      "t_count": 51,
      "t_name": "#happyweekend",
      "t_status": "active",
    },
    Object {
      "id": 19,
      "t_color_code": "#778766",
      "t_count": 50,
      "t_name": "#Mondaymotivation",
      "t_status": "active",
    },
    Object {
      "id": 18,
      "t_color_code": "#975565",
      "t_count": 50,
      "t_name": "#beyourvalentine",
      "t_status": "active",
    },
    Object {
      "id": 28,
      "t_color_code": "#B8860B",
      "t_count": 50,
      "t_name": "#rapper",
      "t_status": "active",
    },
    Object {
      "id": 3,
      "t_color_code": "#525997",
      "t_count": 50,
      "t_name": "#beautytutorials",
      "t_status": "active",
    },
    Object {
      "id": 7,
      "t_color_code": "#958882",
      "t_count": 50,
      "t_name": "#nightout",
      "t_status": "active",
    },
    Object {
      "id": 26,
      "t_color_code": "#D97A64",
      "t_count": 33,
      "t_name": "#painting",
      "t_status": "active",
    },
    Object {
      "id": 207,
      "t_color_code": "#c095de",
      "t_count": 15,
      "t_name": "TESTAG",
      "t_status": "active",
    },
    Object {
      "id": 144,
      "t_color_code": "#566b88",
      "t_count": 12,
      "t_name": "#tag",
      "t_status": "active",
    },
    Object {
      "id": 5,
      "t_color_code": "#664653",
      "t_count": 5,
      "t_name": "#TGIF",
      "t_status": "active",
    },
    Object {
      "id": 145,
      "t_color_code": "#6df2e2",
      "t_count": 5,
      "t_name": "#dinner",
      "t_status": "active",
    },
    Object {
      "id": 30,
      "t_color_code": "#00008B",
      "t_count": 5,
      "t_name": "#beastar",
      "t_status": "active",
    },
    Object {
      "id": 140,
      "t_color_code": "#25ac6c",
      "t_count": 4,
      "t_name": "#Khinyadanar",
      "t_status": "active",
    },
    Object {
      "id": 152,
      "t_color_code": "#cca7af",
      "t_count": 4,
      "t_name": "#phyu",
      "t_status": "active",
    },
    Object {
      "id": 27,
      "t_color_code": "#A9A9A9",
      "t_count": 4,
      "t_name": "#cooking",
      "t_status": "active",
    },
    Object {
      "id": 178,
      "t_color_code": "#18481d",
      "t_count": 4,
      "t_name": "#Cooking",
      "t_status": "active",
    },
    Object {
      "id": 141,
      "t_color_code": "#3520ea",
      "t_count": 3,
      "t_name": "#Ayelinhtike",
      "t_status": "active",
    },
    Object {
      "id": 168,
      "t_color_code": "#6cd3e3",
      "t_count": 3,
      "t_name": "#shooting",
      "t_status": "active",
    },
    Object {
      "id": 154,
      "t_color_code": "#1bd23b",
      "t_count": 3,
      "t_name": "#cosplay",
      "t_status": "active",
    },
    Object {
      "id": 171,
      "t_color_code": "#db5031",
      "t_count": 3,
      "t_name": "#tags",
      "t_status": "active",
    },
    Object {
      "id": 153,
      "t_color_code": "#8bb386",
      "t_count": 2,
      "t_name": "#listen",
      "t_status": "active",
    },
    Object {
      "id": 150,
      "t_color_code": "#e4a8e2",
      "t_count": 2,
      "t_name": "#haha",
      "t_status": "active",
    },
    Object {
      "id": 131,
      "t_color_code": "#2f2e0e",
      "t_count": 2,
      "t_name": "#coffee",
      "t_status": "active",
    },
  ],
  "message": "success",
}
`;

exports[`Hlwint API Service able to modify end time 1`] = `
Object {
  "message": "success",
}
`;

exports[`Hlwint API Service able to retrieve live data 1`] = `
Object {
  "live_lists": Array [
    Object {
      "broadcast_title": "hello",
      "first_name": "yati",
      "id": 70,
      "larger_profile_pic": null,
      "last_name": "zin",
      "location": "OnEarth",
      "nick_name": "yati21",
      "profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/21_18_1770_profile.png",
      "tag": Array [],
      "type": 1,
      "view_count": 32,
    },
    Object {
      "broadcast_title": "tercyduk",
      "first_name": "Reza",
      "id": 3609,
      "larger_profile_pic": null,
      "last_name": "Abdillah",
      "location": "Klaten",
      "nick_name": "reza3609",
      "profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/08_26_363609_profile.png",
      "tag": Array [],
      "type": 1,
      "view_count": 37,
    },
  ],
  "message": "success",
}
`;

exports[`Hlwint API Service able to retrieve live data 2`] = `
Array [
  Object {
    "broadcast_title": "hello",
    "first_name": "yati",
    "id": 70,
    "larger_profile_pic": null,
    "last_name": "zin",
    "location": "OnEarth",
    "nick_name": "yati21",
    "profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/21_18_1770_profile.png",
    "tag": Array [],
    "type": 1,
    "view_count": 32,
  },
  Object {
    "broadcast_title": "tercyduk",
    "first_name": "Reza",
    "id": 3609,
    "larger_profile_pic": null,
    "last_name": "Abdillah",
    "location": "Klaten",
    "nick_name": "reza3609",
    "profile_pic": "https://dashboard.imliveapp.com/public/mlive_img/08_26_363609_profile.png",
    "tag": Array [],
    "type": 1,
    "view_count": 37,
  },
]
`;

exports[`Hlwint API Service able to send gift 1`] = `
Object {
  "message": "success",
}
`;

exports[`Hlwint API Service able to upload last device 1`] = `
Object {
  "message": "success",
  "promotion": "false",
}
`;

exports[`Hlwint API Service ablle to get user following 1`] = `
Object {
  "data": Array [],
  "message": "success",
}
`;

exports[`Hlwint API Service have a correct login service 1`] = `
Object {
  "data": Object {
    "bot": 0,
    "data_usage": null,
    "id": 384,
    "u_account_status": "active",
    "u_biography": null,
    "u_city": "Pathein",
    "u_coins": 13630,
    "u_created_time": "2017-11-13 15:36:22",
    "u_date_of_birth": "null",
    "u_device_os": "android",
    "u_duration": "00:00:00",
    "u_email": "null",
    "u_facebook_id": "367599367030660",
    "u_fan_count": 1,
    "u_first_name": "Ko",
    "u_follower_count": 1,
    "u_full_name": "Ko Kyaw",
    "u_gender": "female",
    "u_google_id": "null",
    "u_income": 0,
    "u_ip_address": "119.2.52.244",
    "u_key_id": "91600",
    "u_larger_profile_pic": null,
    "u_last_device": "android test",
    "u_last_login_time": "2018-01-24 11:09:40",
    "u_last_name": "Kyaw",
    "u_nick_name": "ko384",
    "u_phone_number": "09",
    "u_profile_pic": "https://graph.facebook.com/367599367030660/picture?type=large",
    "u_reason_banned": "",
    "u_reward_status": 0,
    "u_token": "rkQePYzwtsmOwiH6upq1ES1JRWqmnVFShYkSjidpxMrDyReUcxHFAYdKaA0H",
    "u_type": 1,
  },
  "message": "success",
  "version": "18",
}
`;
